app/client/src/components/shared/AddSaveDataWidget.SavePanel.tsx

```diff
@@ -0,0 +1,722 @@
+        layerTypesToIgnore.includes(layer.type)
+      )
+        return;
+
+      if (newSwitches.hasOwnProperty(layer.id)) {
+        newSwitches[layer.id].layer = layer;
+      } else {
+        newSwitches[layer.id] = {
+          id: layer.id,
+          label: layer.title,
+          toggled: layer.visible,
+          requiresFeatureService:
+            layersRequireFeatureService.includes(layer.id) ||
+            fileLayerIds.includes(layer.id),
+          layer: layer,
+        };
+      }
+
+      if (!layersOnMap.includes(layer.id)) layersOnMap.push(layer.id);
+
+      newSwitchesNoLayer[layer.id] = {
+        ...newSwitches[layer.id],
+        layer: null,
+      };
+    });
+
+    Object.keys(newSwitches).forEach((layerId) => {
+      if (layersOnMap.includes(layerId)) return;
+      delete newSwitches[layerId];
+    });
+
+    const oldSwitches: any = {};
+    if (saveLayersList) {
+      Object.keys(saveLayersList).forEach((key) => {
+        const value = saveLayersList[key];
+
+        oldSwitches[key] = {
+          ...value,
+          layer: null,
+        };
+      });
+    }
+
+    if (!deepEqual(newSwitchesNoLayer, oldSwitches)) {
+      setSaveLayersList(newSwitches);
+    }
+  }, [
+    layersInitialized,
+    mapLayerCount,
+    mapView,
+    saveLayersList,
+    setSaveLayersList,
+    visible,
+    widgetLayers,
+  ]);
+
+  const [publishResponse, setPublishResponse] = useState<PublishType>({
+    status: 'idle',
+  });
+
+  if (
+    !mapView ||
+    layerProps.status === 'fetching' ||
+    services.status === 'fetching'
+  ) {
+    return <LoadingSpinner />;
+  }
+
+  async function runPublish(
+    portal: __esri.Portal,
+    serviceMetaData: ServiceMetaDataType,
+    layersToPublish: LayerType[],
+  ) {
+    // check if the name is available in the user's org
+    const nameAvailableResponse = await isServiceNameAvailable(
+      portal,
+      serviceMetaData.label,
+    );
+    if (nameAvailableResponse.error) {
+      setPublishResponse({
+        status: 'failure',
+        error: {
+          message: nameAvailableResponse.error.message,
+        },
+      });
+      return;
+    }
+    if (!nameAvailableResponse.available) {
+      setPublishResponse({
+        status: 'name-not-available',
+      });
+      return;
+    }
+
+    try {
+      const response = await publish({
+        portal,
+        mapView,
+        services,
+        layers: layersToPublish,
+        layerProps,
+        serviceMetaData,
+      });
+
+      // parse the publish output
+      const layerSuccess = !response.layersResult
+        ? true
+        : response.layersResult.success;
+
+      let totalFeatures = 0;
+      let featureFailedCount = 0;
+      response.featuresResult?.forEach((l) => {
+        l.addResults?.forEach((r) => {
+          totalFeatures += 1;
+          if (!r.success) featureFailedCount += 1;
+        });
+      });
+
+      const webMapSuccess = response.webMapResult.success;
```

Comment:
If the _addItem_ endpoint is blocked (or fails), `webMapResult` is undefined and this throws a type error, so might want to guard against that here.
```suggestion
      const webMapSuccess = response.webMapResult?.success ?? false;
```